src/diffusers/pipelines/pipeline_utils.py

```diff
@@ -1239,6 +1256,20 @@ def download(cls, pretrained_model_name, **kwargs) -> Union[str, os.PathLike]:
                 model_info_call_error = e  # save error to reraise it if model is not cached locally
 
         if not local_files_only:
+            filenames = {sibling.rfilename for sibling in info.siblings}
+            model_filenames, variant_filenames = variant_compatible_siblings(filenames, variant=variant)
+            # The variant filenames can have the legacy sharding checkpoint format that we check and throw
+            # a deprecation warning if detected.
+            if len(variant_filenames) > 0 and _check_legacy_sharding_format(filenames, variant):
+                deprecation_message = f"This serialization format is now deprecated to standardize the serialization format between `transformers` and `diffusers`. We recommend you to remove the existing files associated with the current variant ({variant}) and re-obtain them by running a `save_pretrained()`."
```

Comment:
I think it should be a warning, not a deprecation, since it is on pipeline loading